source/core/options.ts

```diff
@@ -1401,7 +1402,7 @@ export default class Options {
 		this._internals.url = url;
 		decodeURI(urlString);
 
-		if (url.protocol === 'unix:') {
+		if (this._internals.enableUnixSockets && url.protocol === 'unix:') {
```

Comment:
Please revert this change. Instead check if this is not enabled inside this if https://github.com/sindresorhus/got/blob/d4c2913ec25968e8d99a4a24e589d3e741072369/source/core/options.ts#L1430 and throw.